### PR TITLE
Add support for TCP (in kubernetes CRD)

### PR DESCRIPTION
**Pull request by mpl**
Also improve integration tests.

### What does this PR do?

This PR adds TCP routing support for the kubernetes CRD provider configuration.

### More

- [X] Added/updated tests
- [X] Added/updated documentation

### Additional Notes

Co-authored-by: Jean-Baptiste Doumenjou <jb.doumenjou@gmail.com>

### CODE REVIEW

**Comment by dtomcej** [comment]
Awesome job on this one!

File: pkg/provider/kubernetes/crd/kubernetes.go [hunk 1]
```diff
@@ -151,6 +151,77 @@ func checkStringQuoteValidity(value string) error {
...
+
+	if !match {
+		return nil, errors.New("service port not found")
+	}
+
+	var servers []config.TCPServer
+	if service.Spec.Type == corev1.ServiceTypeExternalName {
+		servers = append(servers, config.TCPServer{
+			Address: fmt.Sprintf("%s:%d", service.Spec.ExternalName, portSpec.Port),
+			Port:    fmt.Sprintf("%d", portSpec.Port),
```
[the remark this answers is not in the document]
Comment (mpl) [reply pkg/provider/kubernetes/crd/kubernetes.go]: Done

File: pkg/provider/kubernetes/crd/kubernetes.go [hunk 2]
```diff
@@ -151,6 +151,77 @@ func checkStringQuoteValidity(value string) error {
...
+			}
+
+			if port == 0 {
+				return nil, errors.New("cannot define a port")
+			}
+
+			for _, addr := range subset.Addresses {
+				servers = append(servers, config.TCPServer{
+					Address: fmt.Sprintf("%s:%d", addr.IP, port),
+					Port:    fmt.Sprintf("%d", port),
```
[the remark this answers is not in the document]
Comment (mpl) [reply pkg/provider/kubernetes/crd/kubernetes.go]: Done

File: pkg/provider/kubernetes/crd/kubernetes.go [hunk 3]
```diff
@@ -151,6 +151,75 @@ func checkStringQuoteValidity(value string) error {
...
+		if svc.Port == p.Port {
+			portSpec = p
+			match = true
+			break
+		}
+	}
+
+	if !match {
+		return nil, errors.New("service port not found")
+	}
```
Comment (ldez) [inline pkg/provider/kubernetes/crd/kubernetes.go]:
```go
	var portSpec *corev1.ServicePort
	for _, p := range service.Spec.Ports {
		if svc.Port == p.Port {
			portSpec = &p
			break
		}
	}

	if portSpec == nil {
		return nil, errors.New("service port not found")
	}
```

same thing L240

Comment (mpl) [reply pkg/provider/kubernetes/crd/kubernetes.go]: j'ai vu que j'ai loupé ça à cause du git push. j'y reviendrai. enfin, si j'arrive à retrouver le contexte...

**Pull request merged**